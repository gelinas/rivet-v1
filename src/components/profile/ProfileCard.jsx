import React from 'react'

// styles
import './profile.scss'

export default function ProfileCard(props) {
  const {profile} = props

  return (
    <div className='columns is-centered is-mobile'>
      <img
        src={profile.photo ? profile.photo : "https://www.rivet.work/wp-content/uploads/2019/10/Rivet_Logo_UnionBlue-300x138.png"}
        className={'column is-narrow profile_image'}
        alt={`${profile.first_name} ${profile.last_name}`}
      />
      <div className={'column is-narrow profile_details'}>
        <p className='is-size-5 is-size-6-mobile has-text-weight-bold'>
          {`${profile.first_name} ${profile.last_name}`}
        </p>
        <p className='is-size-6 is-size-7-mobile'>
          Email:&nbsp;
          <span className='color_shark'>
            {profile.email}
          </span>
        </p>
        <p className='is-size-6 is-size-7-mobile'>
          Phone:&nbsp;
          <span className='color_shark'>
            {profile.phone}
          </span>
        </p>
        <br />
        <p className='is-size-6 is-size-7-mobile'>
          Address:&nbsp;
        </p>
        <p className='is-size-6 is-size-7-mobile color_shark'>
          {profile.address}
        </p>
        <p className='is-size-6 is-size-7-mobile color_shark'>
          {`${profile.city}, ${profile.state} ${profile.zip}`}
        </p>
      </div>
    </div>
  )
}