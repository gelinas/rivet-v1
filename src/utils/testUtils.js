// testUtils.js rexports @testing-library/react with a customRender method as render
// the customRender method wraps tests in the react-router and redux store providers

import React from 'react';
import { render } from '@testing-library/react'
import { BrowserRouter as Router } from 'react-router-dom';

// redux
import { createStore, applyMiddleware } from 'redux';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import rootReducer from '../reducers';

const store = createStore(rootReducer, applyMiddleware(thunk))

const ProviderWrapper = ({ children }) => {
  return (
    <Provider store={store}>
      <Router>
        {children}
      </Router>
    </Provider>
  )
}

const customRender = (ui, options) =>
  render(ui, { wrapper: ProviderWrapper, ...options })

// re-export everything
export * from '@testing-library/react'

// override render method
export { customRender as render }