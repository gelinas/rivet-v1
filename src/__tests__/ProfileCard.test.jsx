import React from 'react'

// Testing libraries
import { render } from '../utils/testUtils.js'
import '@testing-library/jest-dom/extend-expect'

// Components to test
import ProfileCard from '../components/profile/ProfileCard'

const data = {
  id: 10,
  created_at: "2020-01-03 21:13:44",
  updated_at: "2020-01-03 21:13:44",
  first_name: "Louis",
  last_name: "Gelinas",
  phone: "313-920-1944",
  email: "jlgelinas@gmail.com",
  address: "645 E Bethune",
  city: "Detroit",
  state: "MI",
  zip: "48202",
  photo: "https:\/\/www.louisgelinas.com\/static\/media\/louis-headshot.d09504cd.jpg",
  notes: null
}

describe('Tests for ProfileCard', () => {

  test('should match snapshot', () => {
    const tree = render(
      <ProfileCard profile={data} />
    )
    expect(tree).toMatchSnapshot()
  })

  test('Should display correct data', () => {
    const {getByText} = render(
      <ProfileCard profile={data} />
    )

    expect(getByText(data.email)).toBeInTheDocument()
    expect(getByText(/645 E Bethune/i)).toBeInTheDocument()
  })
})