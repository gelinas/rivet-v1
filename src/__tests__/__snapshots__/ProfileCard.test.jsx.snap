// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for ProfileCard should match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="columns is-centered is-mobile"
      >
        <img
          alt="Louis Gelinas"
          class="column is-narrow profile_image"
          src="https://www.louisgelinas.com/static/media/louis-headshot.d09504cd.jpg"
        />
        <div
          class="column is-narrow profile_details"
        >
          <p
            class="is-size-5 is-size-6-mobile has-text-weight-bold"
          >
            Louis Gelinas
          </p>
          <p
            class="is-size-6 is-size-7-mobile"
          >
            Email: 
            <span
              class="color_shark"
            >
              jlgelinas@gmail.com
            </span>
          </p>
          <p
            class="is-size-6 is-size-7-mobile"
          >
            Phone: 
            <span
              class="color_shark"
            >
              313-920-1944
            </span>
          </p>
          <br />
          <p
            class="is-size-6 is-size-7-mobile"
          >
            Address: 
          </p>
          <p
            class="is-size-6 is-size-7-mobile color_shark"
          >
            645 E Bethune
          </p>
          <p
            class="is-size-6 is-size-7-mobile color_shark"
          >
            Detroit, MI 48202
          </p>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="columns is-centered is-mobile"
    >
      <img
        alt="Louis Gelinas"
        class="column is-narrow profile_image"
        src="https://www.louisgelinas.com/static/media/louis-headshot.d09504cd.jpg"
      />
      <div
        class="column is-narrow profile_details"
      >
        <p
          class="is-size-5 is-size-6-mobile has-text-weight-bold"
        >
          Louis Gelinas
        </p>
        <p
          class="is-size-6 is-size-7-mobile"
        >
          Email: 
          <span
            class="color_shark"
          >
            jlgelinas@gmail.com
          </span>
        </p>
        <p
          class="is-size-6 is-size-7-mobile"
        >
          Phone: 
          <span
            class="color_shark"
          >
            313-920-1944
          </span>
        </p>
        <br />
        <p
          class="is-size-6 is-size-7-mobile"
        >
          Address: 
        </p>
        <p
          class="is-size-6 is-size-7-mobile color_shark"
        >
          645 E Bethune
        </p>
        <p
          class="is-size-6 is-size-7-mobile color_shark"
        >
          Detroit, MI 48202
        </p>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
